dashboard/src/pages/login/CoreConnect.tsx

```diff
@@ -44,6 +45,7 @@ const CoreConnect = () => {
     values: CoreConnectionInfo,
     actions: FormikHelpers<CoreConnectionInfo>
   ) => {
+    setHttpProtocol(values.protocol == "http");
```

Comment:
need to add a special case for localhost/127.0.0.1/etc, which does not need mixed content enabled

https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content#loading_locally_delivered_mixed-resources